pallets/collator-selection/src/migrations.rs

```diff
@@ -17,8 +17,11 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 
 use super::*;
+#[allow(deprecated)]
+use frame_support::migration::remove_storage_prefix;
```

Comment:
wondering if we should allow deprecated code for the whole migrations module.
This is bound to continue occurring in the future on other calls if we don't

Reply:
This is only for `frame_support::migration::remove_storage_prefix` 
For the whole file there needs to be a `!` at the start.

Reply:
right, I'm saying it might be a good idea to do that

Reply:
Aha, I didn't want to accidentally use deprecated code in future migrations, that's why I didn't set it for the whole file.

Reply:
maybe instead of allowing deprecations it'd be better to just comment out the migrations that turn deprecated.

Or - the other alternative - we get rid of the code once it's applied to the chain.
If it becomes painful to maintain, there's no point keeping it

Reply:
I think it's fine for now to keep this as a historical record. If it becomes more painful we can start commenting it out or just delete it outright.